src/usersyncroniser.ts

```diff
@@ -338,7 +345,8 @@ export class UserSyncroniser {
             return UserSyncroniser.ERR_CHANNEL_MEMBER_NOT_FOUND;
         }
         const state = await this.GetUserStateForGuildMember(member, ev.content!.displayname);
-        return this.ApplyStateToRoom(state, roomId, member.guild.id);
+        await this.ApplyStateToRoom(state, roomId, member.guild.id);
+        return "";
```

Comment:
wha? why do we not want to return the result of ApplyStateToRoom? This function must return a IUserState

Reply:
actually it needs to return a string

Reply:
it's odd, because sorus changes here are unrelated to that but tsc started to say "nope can't do this" so soru fixed it here

`ApplyStateToRoom` returns void

should soru add `public static readonly ERR_NO_ERROR = "";` and return that?

Reply:
Sounds fine, yeah